Tiltfile

```diff
@@ -0,0 +1,201 @@
+##############################
+
+# Set up a local_resource build of the Velero binary. The binary is written to _tiltbuild/velero.
+local_resource(
+    "velero_server_binary",
+    cmd = 'cd ' + '.' + ';mkdir -p _tiltbuild;PKG=. BIN=velero GOOS=linux GOARCH=amd64 VERSION=main GIT_TREE_STATE=dirty OUTPUT_DIR=_tiltbuild ./hack/build.sh',
+    deps = ["cmd", "internal", "pkg"],
+    ignore = ["pkg/cmd"],
+)
+
+local_resource(
+    "velero_local_binary",
+    cmd = 'cd ' + '.' + ';mkdir -p _tiltbuild/local;PKG=. BIN=velero GOOS=' + local_goos + ' GOARCH=amd64 VERSION=main GIT_TREE_STATE=dirty OUTPUT_DIR=_tiltbuild/local ./hack/build.sh',
+    deps = ["internal", "pkg/cmd"],
+)
+
+tilt_dockerfile_header = """
+FROM gcr.io/distroless/base:debug as tilt
+WORKDIR /
+COPY --from=tilt-helper /start.sh .
+COPY --from=tilt-helper /restart.sh .
+COPY velero .
+"""
+
+dockerfile_contents = "\n".join([
+    tilt_helper_dockerfile_header,
+    additional_docker_helper_commands,
+    tilt_dockerfile_header,
+    additional_docker_build_commands,
+])
+
+ # Set up an image build for Velero. The live update configuration syncs the output from the local_resource
+ # build into the container.
+docker_build(
+    ref = "velero/velero",
+    context = "_tiltbuild/",
+    dockerfile_contents = dockerfile_contents,
+    target = "tilt",
+    entrypoint=["sh", "/start.sh", "/velero"],
+    live_update=[
+        sync("./_tiltbuild/velero", "/velero"),
+        run("sh /restart.sh"),
+    ])
+
+##############################
+# Setup plugins
+##############################
+
+providers = {}
+
+def load_provider_tiltfiles():
+    provider_repos = settings.get("provider_repos", [])
+
+    for repo in provider_repos:
+        file = repo + "/tilt-provider.json"
+        provider_details = read_json(file, default = {})
+        if type(provider_details) != type([]):
+            provider_details = [provider_details]
+        for item in provider_details:
+            provider_name = item["name"]
+            provider_config = item["config"]
+            if "context" in provider_config:
+                provider_config["context"] = repo + "/" + provider_config["context"]
+            else:
+                provider_config["context"] = repo
+            if "go_main" not in provider_config:
+                provider_config["go_main"] = "main.go"
+            providers[provider_name] = provider_config
+
+# Enable only provider plugins listed in 'enable_providers' in tilt-settings.json (by provider's name)
+def enable_providers():
+    user_enable_providers = settings.get("enable_providers", [])
+    for name in user_enable_providers:
+        enable_provider(name)
+
+# Configures a provider by doing the following:
+#
+# 1. Enables a local_resource go build of the provider's local binary
+# 2. Configures a docker build for the provider, with live updating of the local binary
+def enable_provider(name):
+    p = providers.get(name)
+    plugin_name = p.get("plugin_name")
```

Comment:
Hmm, I'm getting an error when I run this locally with https://github.com/vmware-tanzu/velero-plugin-for-aws/pull/66 checked out as `p.get("plugin_name")` returns `None`. At the point of calling this, the contents of `p` are `{"context": "../velero-plugin-for-aws/.", "image": "velero/velero-plugin-for-aws", "live_reload_deps": ["velero-plugin-for-aws"], "go_main": "./velero-plugin-for-aws"}`. Given that it's being copied into the container, should we be referencing the `go_main` value here?

Reply:
The plugin repo is missing an update to the settings file that adds "plugin_name"! Let me push those and maybe also we can merge them first.

Reply:
Done ✅